ncbi_genome_download/core.py

```diff
@@ -185,56 +252,98 @@ def download(**kwargs):
-    section = kwargs.pop('section', EDefaults.SECTIONS.default)
-    assert section in EDefaults.SECTIONS.choices, "Unsupported section: {}".format(section)
-    group = kwargs.pop('group', EDefaults.TAXONOMIC_GROUPS.default)
-    assert group in EDefaults.TAXONOMIC_GROUPS.choices, "Unsupported group: {}".format(group)
-    if group == 'all':
+    assert args.section in EDefaults.SECTIONS.choices, "Unsupported section: {}".format(args.section)
+    for group in args.group.split(','):
+        assert group in EDefaults.TAXONOMIC_GROUPS.choices, "Unsupported group: {}".format(group)
+    if 'all' in args.group.split(','):
         groups = SUPPORTED_TAXONOMIC_GROUPS
     else:
-        groups = [group, ]
-    uri = kwargs.pop('uri', EDefaults.URI.default)
-    output = kwargs.pop('output', EDefaults.OUTPUT.default)
-    file_format = kwargs.pop('file_format', EDefaults.FORMATS.default)
-    assert file_format in EDefaults.FORMATS.choices, \
-        "Unsupported file format: {}".format(file_format)
-    assembly_level = kwargs.pop('assembly_level', EDefaults.ASSEMBLY_LEVELS.default)
-    assert assembly_level in EDefaults.ASSEMBLY_LEVELS.choices, \
-        "Unsupported assembly level: {}".format(assembly_level)
-    refseq_category = kwargs.pop('refseq_category', EDefaults.REFSEQ_CATEGORIES.default)
-    assert refseq_category in EDefaults.REFSEQ_CATEGORIES.choices, \
-        "Unsupported refseq_category: {}".format(refseq_category)
-    genus = kwargs.pop('genus', EDefaults.GENUS.default)
-    species_taxid = kwargs.pop('species_taxid', EDefaults.SPECIES_TAXID.default)
-    taxid = kwargs.pop('taxid', EDefaults.TAXID.default)
-    human_readable = kwargs.pop('human_readable', False)
-    parallel = kwargs.pop('parallel', EDefaults.NB_PROCESSES.default)
-    table = kwargs.pop('metadata_table', EDefaults.TABLE.default)
-    # FIXME: improve error handling and feedback
-    assert len(kwargs) == 0, "Unrecognized option(s): {}".format(kwargs.keys())
+        groups = args.group.split(',')
 
-    if table is not None:
-        logging.info('Creating metadata file: %r', table)
-        with open(table, 'wt') as metadata_table:
+
+    formats = args.file_format.split(',')
+    for format in formats:
+        assert format in EDefaults.FORMATS.choices, \
+            "Unsupported file format: {file_format}".format(file_format=format)
+    if 'all' in formats:
+        formats = EFormats.keys()
+
+    assert args.assembly_level in EDefaults.ASSEMBLY_LEVELS.choices, \
+        "Unsupported assembly level: {}".format(args.assembly_level)
+    assert args.refseq_category in EDefaults.REFSEQ_CATEGORIES.choices, \
+        "Unsupported refseq_category: {}".format(args.refseq_category)
+
+    if args.metadata_table:
+        logging.info('Creating metadata file: %r', args.metadata_table)
+        with open(args.metadata_table, 'wt') as metadata_table:
             metadata_table.write(get_table_header())
 
+    if bool(args.taxid):
```

Comment:
Any particular reason why you'd be using `bool` here?

Reply:
Part of debugging process, turned out to b code elsewhere. Removed it.